src/state/contexts/TransactionState.tsx

```diff
@@ -1,21 +1,28 @@
 import { Dispatch, createContext, useContext, useReducer } from 'react';
 
+import { TRANSACTION_WORKFLOW_STATUS } from '../../components/layout/TransactionWorkflow/TransactionWorkflow';
 import {
   AntInteraction,
+  ArweaveTransactionID,
   CONTRACT_TYPES,
   ContractType,
   REGISTRY_INTERACTION_TYPES,
   RegistryInteraction,
   TransactionData,
 } from '../../types';
-import { ARNS_REGISTRY_ADDRESS } from '../../utils/constants';
+import {
+  ARNS_REGISTRY_ADDRESS,
+  STUB_ARWEAVE_TXID,
+} from '../../utils/constants';
 import { TransactionAction } from '../reducers/TransactionReducer';
 
 export type TransactionState = {
-  transactionData: TransactionData; // data that will be used to perform the transaction.
+  deployedTransactionId?: ArweaveTransactionID;
+  transactionData: Partial<TransactionData>; // data that will be used to perform the transaction.
   contractType: ContractType;
   interactionType: AntInteraction | RegistryInteraction;
-  workflowStage: 'pending' | 'confirmed' | 'deployed' | 'successful' | 'failed';
+  workflowStage: TRANSACTION_WORKFLOW_STATUS;
+  error?: any;
```

Comment:
nit: not needed - emit any errors to the `eventEmitter` and we'll handle errors globally

Reply:
removed